inference-engine/tests/unit/gna/ngraph/transformations/gna_insert_transpose_before_matmul.cpp

```diff
@@ -0,0 +1,151 @@
+                                              ngraph::ParameterVector{input_params});
+}
+
+TEST(TransformationTests, InsertTransposeBeforeMatmulTestShapeNotSupported) {
+    std::shared_ptr<ngraph::Function> func(nullptr), reference_func(nullptr);
+
+    {
+        func = createFunction(ngraph::PartialShape{2, 9}, ngraph::Shape{9, 2}, ngraph::Shape{2, 1});
+
+        ngraph::pass::Manager m;
+        m.register_pass<ngraph::pass::InitNodeInfo>();
+        m.register_pass<GNAPluginNS::InsertTransposeBeforeMatmul>();
+        m.run_passes(func);
+        ASSERT_NO_THROW(check_rt_info(func));
+    }
+
+    reference_func = createFunction(ngraph::PartialShape{2, 9}, ngraph::Shape{9, 2}, ngraph::Shape{2, 1});
+
+    const FunctionsComparator func_comparator = FunctionsComparator::with_default().enable(FunctionsComparator::ATTRIBUTES);
+    const FunctionsComparator::Result result = func_comparator(func, reference_func);
+    ASSERT_TRUE(result.valid);
+}
+
+TEST(TransformationTests, InsertTransposeBeforeMatmulTestReshapeInOutEq) {
+    std::shared_ptr<ngraph::Function> func(nullptr), reference_func(nullptr);
+    const ngraph::Shape data_shape{9, 2};
+
+    {
+        func = createFunction(ngraph::PartialShape{9, 2}, ngraph::Shape{9, 2}, ngraph::Shape{2, 1});
+
+        ngraph::pass::Manager m;
+        m.register_pass<ngraph::pass::InitNodeInfo>();
+        m.register_pass<GNAPluginNS::InsertTransposeBeforeMatmul>();
+        m.run_passes(func);
+        ASSERT_NO_THROW(check_rt_info(func));
+    }
+
+    reference_func = createFunction(ngraph::PartialShape{9, 2}, ngraph::Shape{9, 2}, ngraph::Shape{2, 1});
+
+    const FunctionsComparator func_comparator = FunctionsComparator::with_default().enable(FunctionsComparator::ATTRIBUTES);
+    const FunctionsComparator::Result result = func_comparator(func, reference_func);
+    ASSERT_TRUE(result.valid);
+}
+
+// ---------------------------------------------------------------------------------------------------------------------
+
+class InsertTransposeBeforeMatmulTestFixture: public CommonTestUtils::TestsCommon,
+                               public ::testing::WithParamInterface<std::tuple<ngraph::PartialShape, ngraph::Shape, ngraph::Shape>> {
+public:
+    void SetUp() override;
+    std::shared_ptr<ngraph::Function> get_initial_function(const ngraph::PartialShape & input_shape,
+                                                   const ngraph::Shape & reshape_shape,
+                                                   const ngraph::Shape & matmul_shape);
+    std::shared_ptr<ngraph::Function> get_reference(const ngraph::PartialShape & input_shape);
+public:
+    std::shared_ptr<ngraph::Function> function, reference_function;
+};
+
+void InsertTransposeBeforeMatmulTestFixture::SetUp() {
+    const auto& input_shape = std::get<0>(GetParam());
+    const auto& reshape_shape = std::get<1>(GetParam());
+    const auto& matmul_shape = std::get<2>(GetParam());
```

Comment:
jfyi, the following code mae be used in future:

const auto& input_shape, reshape_shape, matmul_shape;
std::tie(input_shape , reshape_shape, matmul_shape) = this->GetParam();

Reply:
may be but we will get one long line, which will be less readable

Reply:
error: declaration of ‘const auto& input_shape’ has no initializer

Reply:
may be const auto& [input_shape, reshape_shape, matmul_shape] = this->GetParam(); ?

Reply:
yes, yes, I've just copied your code) but this is just a possible way, I don't mind leaving it as is.